### ISSUE TITLE
Invalid record does not pass invitation_code to rendered view

### ISSUE THREAD

**Issue by baldwindavid** [rymai/devise_invitable#1]
Thanks for your work on this module.  I am currently using the edge branch with Rails 3 RC and Devise 1.1.1.  I seem to be running into an issue where the invitation_token does not survive to the rendered edit action in the case of an invalid password.  

Suppose I require a password to be a length of 8.  If the invited user enters a short password, the invite acceptance will fail and render the edit action.  However, the invitation token is set to nil prior to the attempted save of the object.  The invitation_token is then not available in the rendered action, so subsequent attempts at entering a valid password will yield a missing invitation_token error.  Perhaps I am doing something else wrong, but I fixed this by adding this to the update method in Devise::InvitationsController...

```
self.resource.invitation_token = params[resource_name][:invitation_token]
```

Here is the full method...

```
def update
  self.resource = resource_class.accept_invitation(params[resource_name])

  if resource.errors.empty?
    set_flash_message(:notice, :updated)
    sign_in_and_redirect(resource_name, resource)
  else
    self.resource.invitation_token = params[resource_name][:invitation_token]
    render_with_scope :edit
  end
end
```

Is there something I am missing here?

Best Regards,
David Baldwin

**Comment by rymai** [comment]
Hey David!

Thanks for taking the time to report this issue, it's something I actually didn't test this case.

I fixed it in the edge branch now: 4372312c7e11854b353e452517378bd524c84937

I added a `@password_required` flag in the model that is set to true just before checking `valid?` on the record.

Without this flag, the `password_required?` method returned `false` and I was just checking that a `password` was present for clearing the `invitation_token`, just before trying to save the record.
That `save` then failed because `password_required?` relied on the absence of `invitation_token`.
So the problem was that, with the old implementation, I had to clear the `invitation_token` before saving to be sure validations was performed on the password.
Now, I check that the record is valid before clearing the invitation_token.

Let me know if that's good for you.

Regardd,
Rémy
